File: chromium_src/chrome/browser/ui/browser.cc
/* Copyright (c) 2019 The Brave Authors. All rights reserved.
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this file,
 * You can obtain one at http://mozilla.org/MPL/2.0/. */

#include "brave/browser/ui/brave_browser.h"
#include "brave/browser/ui/brave_tab_strip_model_delegate.h"
#include "brave/browser/ui/tabs/brave_tab_strip_model.h"
#include "chrome/browser/ui/browser_command_controller.h"

#define BrowserTabStripModelDelegate BraveTabStripModelDelegate

#include <chrome/browser/ui/browser.cc>

#undef BrowserTabStripModelDelegate

File: components/speedreader/resources/panel/api/browser.ts
/* Copyright (c) 2023 The Brave Authors. All rights reserved.
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this file,
 * You can obtain one at https://mozilla.org/MPL/2.0/. */

import * as mojom from 'gen/brave/components/speedreader/common/speedreader_toolbar.mojom.m.js'

// Provide access to all the generated types
export * from 'gen/brave/components/speedreader/common/speedreader_toolbar.mojom.m.js'

interface API {
  dataHandler: mojom.ToolbarDataHandlerRemote
  eventsRouter: mojom.ToolbarEventsHandlerCallbackRouter
}

class ToolbarHandlerAPI implements API {
  dataHandler: mojom.ToolbarDataHandlerRemote
  eventsRouter: mojom.ToolbarEventsHandlerCallbackRouter

  constructor() {
    this.dataHandler = new mojom.ToolbarDataHandlerRemote()
    this.eventsRouter = new mojom.ToolbarEventsHandlerCallbackRouter()

    const factory = mojom.ToolbarFactory.getRemote()
    factory.createInterfaces(
      this.dataHandler.$.bindNewPipeAndPassReceiver(),
      this.eventsRouter.$.bindNewPipeAndPassRemote())
  }
}

let apiInstance: API

export default function getToolbarAPI() {
  if (!apiInstance) {
    apiInstance = new ToolbarHandlerAPI()
  }
  return apiInstance
}